sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_models.py

```diff
@@ -437,6 +437,23 @@ def __init__(self, **kwargs):
         self.error = kwargs.get("error", None)
         self.is_error = True
 
+    def __getattr__(self, attr):
+        result_set = set()
+        result_set.update(
+            RecognizeEntitiesResult().keys() + RecognizePiiEntitiesResult().keys()
+            + DetectLanguageResult().keys() + RecognizeLinkedEntitiesResult().keys()
+            + AnalyzeSentimentResult().keys() + ExtractKeyPhrasesResult().keys()
+        )
+        result_attrs = result_set.difference(DocumentError().keys())
+        if attr in result_attrs:
+            raise AttributeError(
+                "The batched result has a DocumentError with the following details. "
+                "Resolve the error or filter for only successful results using the is_error property.\n"
+                "Document Id: {}\nError: {} - {}\n".
+                format(self.id, self.error["inner_error"]["code"], self.error["inner_error"]["message"])
+            )
+        return self.__getattribute__(attr)
```

Comment:
Calling `__getattribute__` from `__getattr__` seems unusual (don't recall ever having seen it before). If you've gotten to `__getattr__`, then some `__getattribute__` in the MRO has already been called, and raised an `AttributeError`, so this would be calling it again, I think. Offhand I would not actually be surprised if this resulted in an infinite recursion, but it clearly doesn't, so evidently Python goes to efforts not to call `__getattr__` from an exception twice. 

Still, TLDR  I think you could just raise the basic `AttributeError` yourself, here, and personally I think that would be simpler/clearer. 

Probably good to solicit more opinions though in case I am off about any details cc @johanste @lmazuel

Reply:
I would call `super`'s `__getattr__` and let that raise the exception (or, possibly handling things appropriately).

Reply:
If `object` is the parent there is no `__getattr__`?

Reply:
Thanks, I will raise the default `AttributeError` myself. Also @bryevdv curious how you feel about the error message? We got UX feedback to "tell me what the problem is, not how to deal with it". Do you think it would be better just to raise something like this, for example:

`AttributeError: 'DocumentError' object has no attribute 'sentiment'`
`Document Id: 0`
`Error: invalidDocument - A document within the request was too large to be processed. Limit document size to: 5120`

Reply:
I personally like this new version better. You might add something like "This is because..." after the "has no attribute" and before the following error details.

Reply:
> If object is the parent there is no __getattr__? You'll get an AttributeError but not the one you want:

Sorry. Yes. Brain not working. Bad suggestion.